cmd/release-controller/info.go

```diff
@@ -30,8 +31,13 @@ type CachingReleaseInfo struct {
 
 func NewCachingReleaseInfo(info ReleaseInfo, size int64) ReleaseInfo {
 	cache := groupcache.NewGroup("release", size, groupcache.GetterFunc(func(ctx groupcache.Context, key string, sink groupcache.Sink) error {
-		parts := strings.Split(key, "\x00")
-		s, err := info.ChangeLog(parts[0], parts[1])
+		var s string
+		var err error
+		if parts := strings.Split(key, "\x00"); len(parts) == 2 {
```

Comment:
You should refactor the key so that it starts with the type of data you're using as a constant `changelog\x00` and `releaseinfo\x00`, then branch based on recognizing that first segment.  Callers should put it in the cache the correct way.

Reply:
Then this would be a switch statement on part[0].